hueplusplus/HueLight.cpp

```diff
@@ -23,217 +26,226 @@
+}
 
-std::string HueLight::getName() {
-  refreshState();
-  return state["name"].asString();
+std::string HueLight::getName()
+{
+    refreshState();
+    return state["name"];
 }
 
-std::string HueLight::getName() const { return state["name"].asString(); }
+std::string HueLight::getName() const
+{
+    return state["name"];
+}
 
-std::string HueLight::getModelId() const { return state["modelid"].asString(); }
+std::string HueLight::getModelId() const
+{
+    return state["modelid"];
+}
 
-std::string HueLight::getUId() const {
-  if (state.isMember("uniqueid")) {
-    return state["uniqueid"].asString();
-  }
-  return std::string();
+std::string HueLight::getUId() const
+{
+    if (state.count("uniqueid"))
+    {
+        return state["uniqueid"];
+    }
+    return std::string();
 }
 
-std::string HueLight::getManufacturername() const {
-  if (state.isMember("manufacturername")) {
-    return state["manufacturername"].asString();
-  }
-  return std::string();
+std::string HueLight::getManufacturername() const
+{
+    if (state.count("manufacturername"))
+    {
+        return state["manufacturername"];
+    }
+    return std::string();
 }
 
-std::string HueLight::getProductname() const {
-  if (state.isMember("productname")) {
-    return state["productname"].asString();
-  }
-  return std::string();
+std::string HueLight::getProductname() const
+{
+    if (state.count("productname"))
+    {
+        return state["productname"];
+    }
+    return std::string();
 }
 
-std::string HueLight::getLuminaireUId() const {
-  if (state.isMember("luminaireuniqueid")) {
-    return state["luminaireuniqueid"].asString();
-  }
-  return std::string();
+std::string HueLight::getLuminaireUId() const
+{
+    if (state.count("luminaireuniqueid"))
+    {
+        return state["luminaireuniqueid"];
+    }
+    return std::string();
 }
 
-std::string HueLight::getSwVersion() {
-  refreshState();
-  return state["swversion"].asString();
+std::string HueLight::getSwVersion()
+{
+    refreshState();
+    return state["swversion"];
 }
 
-std::string HueLight::getSwVersion() const {
-  return state["swversion"].asString();
+std::string HueLight::getSwVersion() const
+{
+    return state["swversion"];
 }
 
-bool HueLight::setName(const std::string &name) {
-  Json::Value request(Json::objectValue);
-  request["name"] = name;
-  Json::Value reply = SendPutRequest(request, "/name");
+bool HueLight::setName(const std::string& name)
+{
+    nlohmann::json request = nlohmann::json::object();
+    request["name"] = name;
+    nlohmann::json reply = SendPutRequest(request, "/name");
 
-  // Check whether request was successful
-  return !reply[0].isNull() && reply[0].isMember("success") &&
-         reply[0]["success"]["/lights/" + std::to_string(id) + "/name"] == name;
+    // Check whether request was successful
+    return reply.size() > 0 && reply[0].count("success")
+        && reply[0]["success"]["/lights/" + std::to_string(id) + "/name"] == name;
```

Comment:
Should there be anotther count check?
```suggestion
        && reply[0]["success"].count("/lights/" + std::to_string(id) + "/name")
        && reply[0]["success"]["/lights/" + std::to_string(id) + "/name"] == name;
```